Slim/Control/Queries.pm

```diff
@@ -4636,25 +4679,32 @@ sub worksQuery {
 		my $sth = $dbh->prepare_cached($sql);
 		$sth->execute( @{$p} );
 
-		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort );
-		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort);
+		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort, $images, $album_ids);
+		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort, \$images, \$album_ids);
 
 		while ( $sth->fetch ) {
-			#$id += 0;
+
+			my $image = (split(/,/, $images))[0];
+
 			utf8::decode($work) if $work;
 			utf8::decode($composer) if $composer;
-			$request->addResultLoop($loopname, $chunkCount, 'work', $work);
 			$request->addResultLoop($loopname, $chunkCount, 'work_id', $workId);
+			if ( $composerCount == 1 && $composerId eq $artistID ) {
+				$request->addResultLoop($loopname, $chunkCount, 'single_composer', 1);
+			}
 			$request->addResultLoop($loopname, $chunkCount, 'composer', $composer);
+			$request->addResultLoop($loopname, $chunkCount, 'work', $work);
 			$request->addResultLoop($loopname, $chunkCount, 'composer_id', $composerId);
-			$request->addResultLoop($loopname, $chunkCount, 'image', "html/images/works.png");
-#			$request->addResultLoop($loopname, $chunkCount, 'year', $year);
-#			$request->addResultLoop($loopname, $chunkCount, 'from_search', $search) if $search;
+			$request->addResultLoop($loopname, $chunkCount, 'image', $image || "html/images/works.png");
```

Comment:
Please use the same method as in the albums query: return the `artwork_track_id` and leave it to the caller to use a fallback image or not.

Reply:
The "works" query can also return a list of images - where one work is on multiple albums. This currently uses an `images` field - I assume, for consistency, that this should now be `artwork_track_ids` ?

Out of curiosity, why is it `artwork_track_id` for albums but `coverid` in most other cases?

Reply:
@mherger is it the fallback or the name you are wanting to change here? Or both?

Reply:
@darrell-k Material will now look for `artwork_track_id` and `artwork_track_ids` in the `works_loop` - I assume this is correct?

Reply:
Yes, correct.

Reply:
OK, changes now pushed. just a reminder: `artwork_track_ids` will contain all artwork ids, not just the first 4.

Reply:
Material was always restricting to 4 anyway. I agree with Michael that this is a UI and not API limit.

Reply:
So do I: it was probably the one thing I didn't argue about ;)

Reply:
> So do I: it was probably the one thing I didn't argue about ;)

I was a little confused about this! :joy: